APMrover2/mode.cpp

```diff
@@ -183,57 +183,26 @@ float Mode::crosstrack_error() const
     if (!is_autopilot_mode()) {
         return 0.0f;
     }
-    return rover.nav_controller->crosstrack_error();
+    return g2.wp_nav.crosstrack_error();
+}
+
+// return desired lateral acceleration
+float Mode::get_desired_lat_accel() const
+{
+    if (!is_autopilot_mode()) {
+        return 0.0f;
+    }
+    return g2.wp_nav.get_lat_accel();
```

Comment:
yes, I agree it's not correct so I've added it to this similar issue. https://github.com/ArduPilot/ardupilot/issues/11284

Note that I don't think it was correct before this PR either.  I fixed a couple of examples of this kind of bad behaviour but this one I only partially fixed.  I say "partially" because now it will at least report zero during pivot turns.  Previously it would have reported a stale lateral acceleration during pivot turns.